### ISSUE TITLE
Create testplan for account freeze unfreeze

### ISSUE THREAD

**Issue by littleskunk** [storj/storj#5307]
[e34826c](https://github.com/storj/storj/commit/e34826c) docs/blueprints: add automatic account freezing/unfreezing blueprint

**Comment by littleskunk** [comment]
My thoughts about some tests we need for sure:
Error message should be something that tells the customer that there is a payment problem and not a usage limit problem.
A customer with a custom limit (increased by support) should get the full custom limit after unfreeze.

**Comment by nadimhq** [comment]
<google-sheets-html-origin><style type="text/css"></style>

Test Scenario | Test Case | Description | Comments
-- | -- | -- | --
Notification | Account Frozen Notification 1 | While the users account is restricted there should be a banner on top of UI notifying the user that their account is frozen because of payment problems |  
  | Insert Funding Notification | Users should be given a notification during the period before their account becomes frozen because of outstanding invoices | the notification should warn them that their account may be frozen until there is a source of funds
  | Account Frozen Notification 2 | If a user tries uploading, downloading, or doing any action tasks- bucket creation, share they should be presented with a notification of their account being frozen and steps to fix this |  
DB | Chore for Outstanding Invoice 1 | There should be a chore that iterates over all paid-tier users and checks for invoices that have not been paid over a specified period of time | what is the set period of time?
  | Chore for Outstanding Invoice 2- Account Frozen Message | If said invoices have not been paid over a specified period of time then the account tied to those invoices are frozen and an message notifying them of this is sent through email |  
  | Chore for Outstanding Invoice 3- Reminder Message | If said invoices have not been paid over a specified period of time but less than the time for it to be considered an outstanding invoice, then the account tied to those invoices should not be frozen however a message notifying them of this being imminent is sent through email |  
  | Frozen Column DB | In this DB column if a users account has the value true then the account usage limit is 0, also if the value is true then this should also be used to display a banner in the UI notifying the user of their restriction on their account likewise if the value is false then the account should not be frozen and this banner should not appear for said user |  
Account Freeze | Limit Download | While account is frozen, user should not be able to download objects |  
  | Limit Upload | While account is frozen, user should not be able to upload objects |  
  | Limit Create Bucket | While account is frozen, user should not be able to create buckets |  
  | Limit Connection to Buckets | User should not be able to use UI or cli tools to interact with or access buckets at all while account is frozen | maybe give an error message in cli when trying to interact with buckets on a frozen account
  | Limit Share | User should not be able to share buckets, objects etc while account is frozen |  
  | Limit Projects | User should not be able to interact with their said projects while account is frozen |  
  | Limit Project Member Usage | Project members should not be able to interact with project owner buckets, objects etc whose account is frozen |  
  | Button on Account Freeze Notification | User should be able to click a link on the notification (which leads to button the billing page) which when pressed attempts to collect payment on a frozen users outstanding invoice | would be nice to have a link that redirects to the part of the billing page containing this new button for quality of life
  | Successful Outstanding Invoice Clear | If user clicks the link or button mentioned previously and there are funds then the account is unfrozen and user should be able to perform tasks download, upload etc |  
  | Unsuccessful Outstanding Invoice Clear | If user clicks the link or button mentioned previously and there are no funds then the account is still frozen and a new error message should pop up notifying user of insufficient funding |  
Account Unfreeze | Restore Custom Limit | If a user had a set custom limit before they were frozen, then when they are unfrozen they should be given back said custom limits f.e project, storage limit, bandwidth limit etc |  
Unpaid User | Over-limit Segment Fees | Currently free tier users are able to use more than whats covered by the free tier coupon (1.65$) by uploading lots of small files, however this would require 10,000 or more segments to even reach over 150gb storage and egress, currently if free tier account has outstanding invoices for over a month then there will be limits placed to 0 | would it make sense to just freeze these accounts and unfreeze just like paid tier accounts once balance is paid off? Since this is done manually at the moment and the original rationale supports this

**Comment by littleskunk** [comment]
I have one question. Should we allow frozen accounts to do list operations? We want them to pay so let's say they lose all data and now they need to restore from backup. If we show them the files they could download after fixing the payment problem they might be more willing to pay.

I would also argue that we should allow delete operations. Let's say the customer is planning to never pay us. The next step for us would be to somehow delete all files (needs to be implemented later). If we allow the customer to delete all files it will reduce the amount of work we have to perform.

For that reason I believe only uploads and download should error out. List and delete should still work.

**Comment by littleskunk** [comment]
Please add one test case for a customer that has no credit card but STORJ token balance. We expect that account to get frozen as well as soon as the balance is all used.
